Helpers/QuoteHelper.php

```diff
@@ -0,0 +1,178 @@
+use Magento\Authorization\Model\UserContextInterface;
+use Magento\Quote\Model\QuoteRepository;
+
+class QuoteHelper
+{
+    /**
+     * @var Logger
+     */
+    private $logger;
+
+    /**
+     * @var Session
+     */
+    private $checkoutSession;
+
+    /**
+     * @var UserContextInterface
+     */
+    private $userContext;
+
+    /**
+     * @var QuoteRepository
+     */
+    private $quoteRepository;
+
+    /**
+     * @param Logger $logger
+     * @param UserContextInterface $userContext
+     * @param QuoteRepository $quoteRepository
+     * @param Session $checkoutSession
+     */
+    public function __construct(
+        Logger $logger,
+        UserContextInterface $userContext,
+        QuoteRepository $quoteRepository,
+        Session $checkoutSession
+    )
+    {
+        $this->logger = $logger;
+        $this->userContext = $userContext;
+        $this->quoteRepository = $quoteRepository;
+        $this->checkoutSession = $checkoutSession;
+    }
+
+    /**
+     * @param null $cartId
+     * @return \Magento\Quote\Model\Quote|null
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getQuote($cartId=null):?\Magento\Quote\Model\Quote
+    {
+        $quote = null;
+        $quoteById = $this->getQuoteById($cartId);
+        if (isset($quoteById)){
+            $this->logger->info('Cart id define quote',[]);
+            return $quoteById;
+        }
+        $contextUserQuote = $this->getQuoteByContextUserId();
+        if(isset($contextUserQuote)){
+            $this->logger->info('Is Quote Context User',[]);
+            return $contextUserQuote;
+        }
+
+        $sessionQuote = $this->getQuoteFromSession();
+        if (isset($sessionQuote)){
+            $this->logger->info('Is Session Quote',[]);
+            return $sessionQuote;
+        }
+        return $quote;
+    }
+
+    /**
+     * Get quote from session if is define
+     * @return \Magento\Quote\Model\Quote|null
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function getQuoteFromSession():?\Magento\Quote\Model\Quote
+    {
+        $quote = null;
+        $quoteId = $this->getQuoteIdFromSession();
+        if($quoteId != null){
+            $quote = $this->checkoutSession->getQuote();
+        }
+        return $quote;
+    }
+
+
+    /**
+     * @return \Magento\Quote\Model\Quote|null
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getQuoteByContextUserId():?\Magento\Quote\Model\Quote
+    {
+        $quote = null;
+        $customerUserId = $this->getContextUserId();
+        if($customerUserId > 0){
+            $quote = $this->quoteRepository->getActiveForCustomer($customerUserId);
+        }
+        return $quote;
+    }
+
+    /**
+     * Load quote with cartID
+     * @param $cartId
+     * @return \Magento\Quote\Model\Quote\Interceptor|null
```

Comment:
replace full qualifier with import (this is for all the place where you write full namespace instead the classname only)

`use \Magento\Quote\Model\Quote\Interceptor;`

then

`@return Interceptor|null`

---

I don't find where `Interceptor` is returned.

As I can see, the repository return a `Quote` or a `CartInterface`